components/org.wso2.carbon.identity.organization.resource.hierarchy.traverse.service/src/main/java/org/wso2/carbon/identity/organization/resource/hierarchy/traverse/service/internal/OrgResourceHierarchyTraverseServiceDataHolder.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.organization.resource.hierarchy.traverse.service.internal;
+
+import org.wso2.carbon.identity.application.mgt.ApplicationManagementService;
+import org.wso2.carbon.identity.organization.management.service.OrganizationManager;
+
+/**
+ * Singleton class that serves as a centralized data holder for key service instances used in the organization
+ * resource hierarchy traversal process.
+ */
+public class OrgResourceHierarchyTraverseServiceDataHolder {
+
+    private static final OrgResourceHierarchyTraverseServiceDataHolder instance =
```

Comment:
```suggestion
    private static final OrgResourceHierarchyTraverseServiceDataHolder INSTANCE =
```
since it is static final constant

Reply:
Addressed with https://github.com/wso2-extensions/identity-organization-management/pull/415/commits/938dc308459c20feb1fa02b2116a22ba884a80be